src/player/player.rs

```diff
@@ -137,6 +140,11 @@ impl SpotifyPlayer {
                     .load(track, true, 0);
                 Ok(())
             }
+            Command::PlayerPreload(track) => {
+                let player = player.as_mut().ok_or(SpotifyError::PlayerNotReady)?;
```

Comment:
```suggestion
let player = self.player.as_mut().ok_or(SpotifyError::PlayerNotReady)?;
```

Reply:
Yeah that makes sense. I skimmed it over because the code looked correct lol. Anyhow, I will look more carefully in the future :)

Reply:
oh no no it was okay! its just that I had pushed to develop in the meantime and things needed adjusting, but it was all good!